File: 7_kyu/Consecutive_Vowels_in_a_String.py
# Tupel of vowels
VOWELS = ('a', 'e', 'i', 'o', 'u')

# Count the number of vwels in order.
def get_the_vowels(word: str) -> int:
    counter = 0
    vowel_index = 0
    for c in word:
        if c == VOWELS[vowel_index]:
            counter += 1
            vowel_index += 1
            if vowel_index == len(VOWELS): vowel_index = 0
    return counter

File: 7_kyu/Consecutive_Vowels_in_a_String.go
package kata

// Array of all vowels, ordered.
// Used as constant. But it isn't const for Go.
var vowels = []rune{'a', 'e', 'i', 'o', 'u'}

// Count the number of vowels in order.
func GetTheVowels(word string) int {
	// counter of vowels found in the correct order.
	counter := 0
	// index of the next vowel in the vowels array.
	// Only this char is counted, all other are ignored.
	vowel_index := 0
	// Iterate over runes
	for _, r := range word {
		if r == vowels[vowel_index] {
			counter++
			vowel_index++
			if vowel_index == len(vowels) {
				vowel_index = 0
			}
		}
	}
	return counter
}
